cli/azd/pkg/containerapps/container_app.go

```diff
@@ -124,6 +131,21 @@ func (cas *containerAppService) DeployYaml(
 		return fmt.Errorf("decoding yaml: %w", err)
 	}
 
+	if shouldPersist := cas.alphaFeatureManager.IsEnabled(persistCustomDomainsFeature); shouldPersist {
+		aca, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName)
+		if err == nil {
```

Comment:
If err was non nil but also was not a 404, do we want to fail the entire operation?

Reply:
the sdk client will retry 3 times for:

```
		o.StatusCodes = []int{
			http.StatusRequestTimeout,      // 408
			http.StatusTooManyRequests,     // 429
			http.StatusInternalServerError, // 500
			http.StatusBadGateway,          // 502
			http.StatusServiceUnavailable,  // 503
			http.StatusGatewayTimeout,      // 504
		}
```

If after that we keep getting errors, most likely all Azure might be down?